### ISSUE TITLE
Incorrect(?) behavior of dartify

### ISSUE THREAD

**Issue by TheBosZ** [pulyaevskiy/node-interop#52]
In Javascript, it's possible to have an object with numeric keys:

```javascript
var obj = {
  'Hello': 'World'
};
obj[1] = 'Hello';
console.log(obj);
```

The interesting behavior is that the numeric key is available with both a raw number and as the number as a string:

```javascript
console.log(obj[1]) > "Hello"
console.log(obj['1']) > "Hello"
```

The dartify helper uses the JS Object.keys which will return only keys as strings. Which means the object that gets created will only ever have String keys.

This threw me for a loop because I was expected my object to have numeric keys.

Is this behavior expected?

I'm having to write a lot of `.toString()` to ensure I'm dealing only in strings.

**Comment by pulyaevskiy** [comment]
Yeah, I'd say that current behavior is as it was intended originally.

As I understand the fact that JS allows accessing a property with it's original key type is simply a "convenience" for the developer. In my experiments regardless of what type I use for keys it gets converted into a string internally. E.g.:

```js
var obj = {};
obj[1] = 'abc';
obj[true] = 'def';
console.log(obj); // {1: "abc", true: "def"}

obj['1'] = 'foo'; // setting string "1" overwrites previously set int key
console.log(obj); // {1: "foo", true: "def"}

obj['true'] = 'bar'; // same with boolean keys
console.log(obj); // {1: "foo", true: "bar"}
```

I tried to find a way to get original key type in JS but was unsuccessful. Iterating in a `for` loop also returns keys as strings.

If there is no way for us to get the key type then it seems unlikely we can do anything here.

One workaround might be to have a proxy class which performs `toString` conversion on each key access. E.g.:

```dart
class JsMap {
  final Map<String, dynamic> _data;
  JsMap(Object jsObject): _data = dartify(jsObject) as Map<String, dynamic>;

  dynamic operator [](dynamic key) {
    return _data['$key'];
  }
}
```

Though a more Dart-like way would be to simply rely on the type system and just use `Map<String, dynamic>` in this case.

**Comment by TheBosZ** [comment]
Feel free to close this. Although a bit of documentation would be nice.

**Comment by pulyaevskiy** [comment]
Just published `1.0.2` which clarifies this behavior.

Thanks for your feedback!
